mitreattack/navlayers/generators/usage_generator.py

```diff
@@ -0,0 +1,199 @@
+            ])
+        elif obj_type == "mitigation":
+            mr = self.get_mitigation(match_pattern)
+            relations = self.source_handle.relationships(mr.id, 'mitigates', source_only=True)
+            out = self.source_handle.query([
+                Filter('type', '=', 'attack-pattern'),
+                Filter('id', 'in', [r.target_ref for r in relations])
+            ])
+        return self.remove_revoked(out)
+
+    def generate_technique_data(self, raw_matches):
+        """
+        Generate technique list of dictionary objects (dictionary form of technique listing for a layer)
+        :param raw_matches: matching attack-pattern objects
+        :return: list of dictionary objects for every technique: score=0 if not in raw_matches, 1 otherwise,
+                    description in comments
+        """
+        shortlist = []
+        for match in raw_matches:
+            xid = ''
+            xphase = ''
+            for ref in match.external_references:
+                if ref.source_name == 'mitre-attack':
+                    xid = ref.external_id
+            for phase in match.kill_chain_phases:
+                if phase.kill_chain_name == 'mitre-attack':
+                    xphase = phase.phase_name
+            shortlist.append((xid, xphase, match.description))
+        full_matrix_listing = self.matrix_handle.get_matrix(self.domain)
+        construct = list()
+        for tactic in full_matrix_listing:
+            for tech in tactic.techniques:
+                construct.append(dict(techniqueID=tech.id, score=0,
+                                      tactic=self.matrix_handle.convert(tactic.tactic.name)))
+            for tech_key in tactic.subtechniques:
+                for subtech in tactic.subtechniques[tech_key]:
+                    construct.append(dict(techniqueID=subtech.id, score=0,
+                                          tactic=self.matrix_handle.convert(tactic.tactic.name)))
+        for entry in shortlist:
+            for tac in construct:
+                if entry[0] == tac['techniqueID'] and (entry[1] == '' or entry[1] == tac['tactic']):
+                    tac['score'] = 1
+                    tac['comment'] = entry[2]
+        return construct
+
+    def generate_layer(self, match, obj_type):
+        """
+        Generate a layer
+        :param match: the pattern to match
+        :param obj_type: the type of object match is (group, software, or mitigation)
+        :return: layer object with annotated techniques
+        """
+
+        typeChecker(type(self).__name__, obj_type, str, "type")
+        categoryChecker(type(self).__name__, obj_type, ["group", "software", "mitigation"], "type")
+        typeChecker(type(self).__name__, match, str, "match")
+        raw_data = self.get_matrix_data(match, obj_type)
+        processed_listing = self.generate_technique_data(raw_data)
+        raw_layer = dict(name=f"AutoGenerated Layer ({match})", domain='enterprise-attack')
```

Comment:
Should follow [layer name convention from attack-website](https://github.com/mitre-attack/attack-website/blob/master/modules/util/buildhelpers.py#L598), e.g `ifconfig (S0101)`.

Reply:
[Should also include a sensible description](https://github.com/mitre-attack/attack-website/blob/master/modules/util/buildhelpers.py#L600-L601).

Reply:
Got it.